src/storage/patch/SparqlUpdatePatchHandler.ts

```diff
@@ -101,9 +99,9 @@ export class SparqlUpdatePatchHandler extends ConvertingPatchHandler {
       this.logger.warn('GRAPH statement in INSERT clause');
       throw new NotImplementedHttpError('GRAPH statements are not supported');
     }
-    if (!(typeof op.where === 'undefined' || this.isBasicGraphPatternWithoutVariables(op.where))) {
-      this.logger.warn('WHERE statements are not supported');
-      throw new NotImplementedHttpError('WHERE statements are not supported');
+    if (!(typeof op.where === 'undefined' || op.where.type === Algebra.types.BGP)) {
```

Comment:
Since it's possible to support other query types as well, why not expose a config option to just allow this? (perhaps via a separate PR?)

Reply:
Good point for the future; we want to avoid applications depending on this though as it's not cross-implementation.

Reply:
(Also begs the question of how to advertise that functionality.)

Reply:
> we want to avoid applications depending on this though as it's not cross-implementation.

Ah yes, of course. The spec should be clear that client MUST NOT assume all servers have support for this, but that they MAY/SHOULD make use of it when they discover it.

> (Also begs the question of how to advertise that functionality.)

Something like a profile for SPARQL Update media type in the `Accept-Patch` header may work here (but I would guess this has been suggested already in the Solid spec discussions).